test/lint/lint-logs.py

```diff
@@ -16,12 +16,12 @@
 
 
 def main():
-    logs_list = check_output(["git", "grep", "--extended-regexp", r"LogPrintf?\(", "--", "*.cpp"], universal_newlines=True, encoding="utf8").splitlines()
+    logs_list = check_output(["git", "grep", "--extended-regexp", r"LogPrintLevel|LogPrintf?\(", "--", "*.cpp"], universal_newlines=True, encoding="utf8").splitlines()
```

Comment:
Does the `|` in this regexp work without `()` context? (I don't see how)

Reply:
It's working well for me wih Python 3.10.4 (thanks for reviewing!) Happy to update with any suggestion.

```
$ test/lint/lint-logs.py 
All calls to LogPrintf(), LogPrint(), and LogPrintLevel() should be terminated with "\n".

src/net.cpp:2408:        LogPrintLevel(BCLog::Level::Error, BCLog::NET, "%s", strError.original);
src/net.cpp:2416:        LogPrintf("%s", strError.original);
src/net.cpp:2425:            LogPrintf("%s", strError.original);
src/net.cpp:2432:            LogPrintf("%s", strError.original);
src/net.cpp:2453:        LogPrintLevel(BCLog::Level::Error, BCLog::NET, "%s", strError.original);
src/netbase.cpp:573:                LogPrintf("wait for connect to %s failed: %s",
src/netbase.cpp:578:                LogPrint(BCLog::NET, "connection attempt to %s timed out", addrConnect.ToString());
src/netbase.cpp:590:                LogPrintf("getsockopt() for %s failed: %s", addrConnect.ToString(), NetworkErrorString(WSAGetLastError()));
```

Reply:
A larger test and with WalletLogPrintf:

```
$ test/lint/lint-logs.py 
All calls to LogPrintf(), LogPrint(), and LogPrintLevel() should be terminated with "\n".

src/addrdb.cpp:147:        LogPrintf("banlist.dat ignored because it can only be read by " PACKAGE_NAME " version 22.x. Remove %s to silence this warning.", fs::quoted(fs::PathToString(m_banlist_dat)));
src/addrman.cpp:388:        LogPrint(BCLog::ADDRMAN, "addrman lost %i new and %i tried addresses due to collisions or invalid addresses", nLostUnk, nLost);
src/banman.cpp:38:        LogPrint(BCLog::NET, "Loaded %d banned node addresses/subnets  %dms", m_banned.size(),
src/banman.cpp:41:        LogPrintf("Recreating the banlist database");
src/banman.cpp:66:    LogPrint(BCLog::NET, "Flushed %d banned node addresses/subnets to disk  %dms", banmap.size(),
src/banman.cpp:194:            LogPrint(BCLog::NET, "Removed banned node address/subnet: %s", sub_net.ToString());
src/net.cpp:2092:                LogPrint(BCLog::NET, "Trying to make an anchor connection to %s", addrConnect.ToString());
src/net.cpp:2408:        LogPrintLevel(BCLog::Level::Error, BCLog::NET, "%s", strError.original);
src/net.cpp:2416:        LogPrintf("%s", strError.original);
src/net.cpp:2425:            LogPrintf("%s", strError.original);
src/net.cpp:2432:            LogPrintf("%s", strError.original);
src/net.cpp:2453:        LogPrintLevel(BCLog::Level::Error, BCLog::NET, "%s", strError.original);
src/netbase.cpp:573:                LogPrintf("wait for connect to %s failed: %s",
src/netbase.cpp:578:                LogPrint(BCLog::NET, "connection attempt to %s timed out", addrConnect.ToString());
src/netbase.cpp:590:                LogPrintf("getsockopt() for %s failed: %s", addrConnect.ToString(), NetworkErrorString(WSAGetLastError()));
src/wallet/wallet.cpp:186:    wallet->WalletLogPrintf("Releasing wallet");
src/wallet/wallet.cpp:1809:        WalletLogPrintf("Rescan completed in %15dms", duration_milliseconds.count());
```
Maybe we should mention WalletLogPrintf too.

Reply:
Yes but I mean I'd have expected `(LogPrintLevel|LogPrintf?)` otherwise it's not clear to the regexp parser what the outer boundary of the group is?

Reply:
Thank you! Tested and updated.